### PR TITLE
Remove BC layers related to new methods and new parameters

### PR DESCRIPTION
**Pull request by nicolas-grekas**
| Q             | A
| ------------- | ---
| Branch?       | 7.0
| Bug fix?      | no
| New feature?  | no
| Deprecations? | no
| Tickets       | -
| License       | MIT
| Doc PR        | -

 * Add method `__toString()` to `InputInterface`
 * Add argument `$id` and `$asGhostObject` to `DumperInterface::isProxyCandidate()` and `getProxyCode()`
 * Add argument `$source` to `FileLoader::registerClasses()`
 * Add argument `$lock` to `Filesystem::appendToFile()`
 * Remove command `translation:update`, use `translation:extract` instead
 * Add argument `$reflector` to `ArgumentResolverInterface::getArguments()` and `ArgumentMetadataFactoryInterface::createArgumentMetadata()`
 * Add method `isNullSafe()` to `PropertyPathInterface`
 * Add argument `$routeParameters` to `UrlMatcher::handleRouteRequirements()`
 * Add argument `$badgeFqcn` to `Passport::addBadge()`
 * Add argument `$lifetime` to `LoginLinkHandlerInterface::createLoginLink()`
 * Add method `getSupportedTypes()` to `DenormalizerInterface` and `NormalizerInterface`
 * Add methods `getConstraint()`, `getCause()` and `__toString()` to `ConstraintViolationInterface`
 * Add method `__toString()` to `ConstraintViolationListInterface`
 * Add method `disableTranslation()` to `ConstraintViolationBuilderInterface`
 * Add argument `$label` to `VarDumper::dump()`

### CODE REVIEW

File: src/Symfony/Component/Console/Input/InputInterface.php [hunk 1]
```diff
@@ -147,4 +144,11 @@ public function isInteractive(): bool;
      * @return void
      */
     public function setInteractive(bool $interactive);
+
+    /**
+     * Returns a stringified representation of the args passed to the command.
+     *
+     * InputArguments MUST be escaped as well as the InputOption values passed to the command.
+     */
+    public function __toString();
```
Comment (94noni) [inline src/Symfony/Component/Console/Input/InputInterface.php]:
```suggestion
    public function __toString(): string;
```

can this be added directly? Or it must be defined in v6 before?

Comment (nicolas-grekas) [reply src/Symfony/Component/Console/Input/InputInterface.php]: Makes zero difference, that's strictly the same for the engine

Comment (94noni) [reply src/Symfony/Component/Console/Input/InputInterface.php]: Oki copy

**Review by javiereguiluz — APPROVED** [review]

File: src/Symfony/Component/Console/Input/InputInterface.php (hunk 1 above)
Comment (wouterj) [reply src/Symfony/Component/Console/Input/InputInterface.php]: I would still add it for consistency (we've also added it to all other `__toString()` definitions in the codebase)

**Review by stof — APPROVED** [review]

**Pull request merged**